api/grpc/interceptor/AuthnInterceptor_test.go

```diff
@@ -19,87 +19,144 @@ import (
 )
 
 const (
-	validIssuer    = "example.com/issuer"
-	validAudience  = "example.com"
-	minimumScope   = "openid"
-	defaultSubject = "u1"
-	testKID        = "test-kid"
+	validIssuer1    = "example.com/issuer"
+	validAudience1  = "example.com"
+	defaultSubject1 = "u1"
+
+	validIssuer2    = "classy.com/issuer"
+	validAudience2  = "classy.com"
+	defaultSubject2 = "u2"
+
+	minimumScope = "openid"
+	testKID      = "test-kid"
 )
 
 func TestInterceptorHoldsValuesFromDiscoveryEndpoint(t *testing.T) {
-	interceptor := createInterceptor()
+	interceptor := AuthnInterceptor{[]*authnProvider{createAuthnProvider1()}}
+
+	result, err := interceptor.validateTokenAndExtractSubject(createToken(createDefaultTokenBuilder1(), tokenSigningKey1))
+
+	assert.NoError(t, err)
+	assert.Equal(t, defaultSubject1, result.SubjectID)
+}
+
+func TestInterceptorHoldsValuesFromSecondDiscoveryEndpoint(t *testing.T) {
+	interceptor := AuthnInterceptor{[]*authnProvider{
+		createAuthnProvider1(), createAuthnProvider2(),
+	}}
+
+	result, err := interceptor.validateTokenAndExtractSubject(createToken(createDefaultTokenBuilder2(), tokenSigningKey2))
+
+	assert.NoError(t, err)
+	assert.Equal(t, defaultSubject2, result.SubjectID)
+}
+
+func TestInterceptorHoldsValuesFromFirstDiscoveryEndpoint(t *testing.T) {
+	interceptor := AuthnInterceptor{[]*authnProvider{
+		createAuthnProvider1(), createAuthnProvider2(),
+	}}
+
+	result, err := interceptor.validateTokenAndExtractSubject(createToken(createDefaultTokenBuilder1(), tokenSigningKey1))
+
+	assert.NoError(t, err)
+	assert.Equal(t, defaultSubject1, result.SubjectID)
+}
+
+func TestAllOkWhen2SameProviders(t *testing.T) {
```

Comment:
... oh ok, you thought about that. I'll resolve my comments ;D

Reply:
On reflection, I think I should create AuthnInterceptor with NewAuthnInterceptor in the tests, to allow for easier reimplementation of the configs -> authnprovider logic.